internal/controllers/machineset/machineset_controller.go

```diff
@@ -415,30 +410,32 @@ func (r *Reconciler) syncReplicas(ctx context.Context, ms *clusterv1.MachineSet,
 			})
 			if err != nil {
 				conditions.MarkFalse(ms, clusterv1.MachinesCreatedCondition, clusterv1.InfrastructureTemplateCloningFailedReason, clusterv1.ConditionSeverityError, err.Error())
-				return errors.Wrapf(err, "failed to clone infrastructure configuration for MachineSet %q in namespace %q", ms.Name, ms.Namespace)
+				return errors.Wrapf(err, "failed to clone infrastructure machine from %s while creating a machine", klog.KRef(machine.Spec.InfrastructureRef.Namespace, machine.Spec.InfrastructureRef.Name))
 			}
+			log = log.WithValues(util.LowerCamelCase(infraRef.Kind), klog.KRef(infraRef.Namespace, infraRef.Name))
 			machine.Spec.InfrastructureRef = *infraRef
 
 			if err := r.Client.Create(ctx, machine); err != nil {
-				log.Error(err, "Unable to create Machine")
+				log := log.WithValues("machine", klog.KObj(machine))
```

Comment:
Can we add the Machine at the beginning of the loop to the logger (l.368)? Just seems a lot more straightforward and we don't have to be careful that we e.g. add it again in l.438

Also safer if we add new log statements over time

Reply:
This cannot done before because machines uses GeneratedName, and so the actual name of the machine is set only after create